source/simulator/helper_functions.cc

```diff
@@ -1320,6 +1323,106 @@ namespace aspect
+                               "but the provided material model reports to be compressible. "
+                               "Please check the consistency of your material model and selected formulation."));
+      }
+    else if (parameters.formulation_mass_conservation == Parameters<dim>::FormulationMassConservation::isothermal_compression
+             || parameters.formulation_mass_conservation == Parameters<dim>::FormulationMassConservation::reference_density_profile
+             || parameters.formulation_mass_conservation == Parameters<dim>::FormulationMassConservation::implicit_reference_density_profile)
+      {
+        AssertThrow(material_model->is_compressible() == true,
+                    ExcMessage("ASPECT detected an inconsistency in the provided input file. "
+                               "The mass conservation equation was selected to be compressible, "
+                               "but the provided material model reports to be incompressible. "
+                               "Please check the consistency of your material model and selected formulation."));
+      }
+
+    // Ensure that the correct heating terms have been selected for the chosen combined formulation
+    // Note that if the combined formulation is 'custom' there is no check
+    // (useful e.g. for smaller scale lithospheric models with shear heating but without adiabatic heating)
+    if (parameters.combined_formulation == Parameters<dim>::CombinedFormulation::isothermal_compression)
+      {
+        AssertThrow(heating_model_manager.adiabatic_heating_enabled(),
+                    ExcMessage("ASPECT detected an inconsistency in the provided input file. "
+                               "The 'isothermal compression' formulation expects adiabatic heating to be enabled, "
+                               "but the heating model manager does not find a heating plugin "
+                               "named 'adiabatic heating' in the list of active plugins. "
+                               "Please check the consistency of your input file."));
+
+        AssertThrow(heating_model_manager.shear_heating_enabled(),
+                    ExcMessage("ASPECT detected an inconsistency in the provided input file. "
+                               "The 'isothermal compression' formulation expects shear heating to be enabled, "
+                               "but the heating model manager does not find a heating plugin "
+                               "named 'shear heating' in the list of active plugins. "
+                               "Please check the consistency of your input file."));
+      }
+    else if (parameters.combined_formulation == Parameters<dim>::CombinedFormulation::BA)
+      {
+        AssertThrow(!heating_model_manager.adiabatic_heating_enabled(),
+                    ExcMessage("ASPECT detected an inconsistency in the provided input file. "
+                               "The BA formulation expects adiabatic heating to be disabled, "
```

Comment:
here and below, maybe expand the abbreviation or say "Boussinesq Approximation (BA)"? Same for the following blocks.